var/spack/repos/builtin/packages/flecsi/package.py

```diff
@@ -57,53 +50,16 @@ class Flecsi(CMakePackage, CudaPackage, ROCmPackage):
     variant("kokkos", default=False, description="Enable Kokkos Support")
     variant("openmp", default=False, description="Enable OpenMP Support")
 
-    # legacy variants
-    variant("coverage", default=False, description="Enable coverage build", when="@:1")
-    variant(
-        "debug_backend", default=False, description="Build Backend with Debug Mode", when="@:1"
-    )
-    variant("disable_metis", default=False, description="Disable FindPackageMetis", when="@:1")
-    variant("doxygen", default=False, description="Enable doxygen", when="@:1")
-    variant("tutorial", default=False, description="Build FleCSI Tutorials", when="@:1")
-    variant("flecstan", default=False, description="Build FleCSI Static Analyzer", when="@:1")
-    variant("external_cinch", default=False, description="Enable External Cinch", when="@:1")
-    variant("unit_tests", default=False, description="Build with Unit Tests Enabled", when="@:1")
-
     # All Current FleCSI Releases
     for level in ("low", "medium", "high"):
-        depends_on("caliper@2.0.1~adiak~libdw", when="@:1 caliper_detail=%s" % level)
-        depends_on("caliper", when="@2.0: caliper_detail=%s" % level)
-        conflicts("^caliper@2.6", when="@2.0: caliper_detail=%s" % level)
-        conflicts("^caliper@2.7", when="@2.0: caliper_detail=%s" % level)
+        depends_on("caliper@2.8:", when=f"@2.0: caliper_detail={level}")
 
     depends_on("graphviz", when="+graphviz")
     depends_on("hdf5+hl+mpi", when="+hdf5")
     depends_on("metis@5.1.0:")
     depends_on("parmetis@4.0.3:")
     depends_on("boost@1.70.0: cxxstd=17 +program_options +stacktrace")
 
-    # FleCSI@1.x
-    depends_on("cmake@3.12:", when="@:1")
-    # Requires cinch > 1.0 due to cinchlog installation issue
-    depends_on("cinch@1.01:", type="build", when="+external_cinch @:1")
-    depends_on("mpi", when="backend=mpi @:1")
-    depends_on("mpi", when="backend=legion @:1")
-    depends_on("mpi", when="backend=hpx @:1")
-    depends_on("legion+shared", when="backend=legion @:1")
-    depends_on("legion+hdf5", when="backend=legion +hdf5 @:1")
-    depends_on("legion build_type=Debug", when="backend=legion +debug_backend")
-    depends_on("legion@cr-20191217", when="backend=legion @:1")
-    depends_on("hpx@1.4.1 cxxstd=17 malloc=system max_cpu_count=128", when="backend=hpx @:1")
-    depends_on("hpx build_type=Debug", when="backend=hpx +debug_backend")
-    depends_on("googletest@1.8.1+gmock", when="@:1")
-    depends_on("python@3.0:", when="+tutorial @:1")
-    depends_on("doxygen", when="+doxygen @:1")
-    depends_on("llvm", when="+flecstan @:1")
-    depends_on("pfunit@3.0:3", when="@:1")
-    depends_on("py-gcovr", when="+coverage @:1")
-    depends_on("openmpi+legacylaunchers", when="+unit_tests ^[virtuals=mpi] openmpi")
-
-    # FleCSI@2.x
     depends_on("cmake@3.15:", when="@2.0:")
```

Comment:
There are many version simplifications we can now make: this can be removed entirely (including in constructs like `+kokkos +rocm @2.0:` and `@2: backend=hpx`), and `@2.0:2.2.1` can become `@:2.2.1`.  We could also consider combining this stanza and the one above (at least removing the blank and possibly re-sorting them to put `boost` together and `cmake` first and so on.